File: daily/python/Dec-6-Calculate-money-in-leetcode-bank.py
class Solution:
    def totalMoney(self, n: int) -> int:
        res = 0
        weekn = n // 7
        day = n % 7
        money = weekn + 1
        j = 0
        while j < day:
            res += money
            money += 1
            j += 1

        if weekn > 0:
            i = 1
            while i <= weekn:
                res += self.sumSeven(i)
                i += 1

        return res

    def sumSeven(self, s: int) -> int:
        sum = 0
        i = 1
        while i <= 7:
            sum += s
            s += 1
            i += 1
        return sum

File: daily/cpp/Dec-6-Calculate-money-in-leetcode-bank.cpp
using namespace std;

class Solution {
public:
  int totalMoney(int n) {
    int res = 0;
    int weekn = n / 7;
    int day = n % 7;
    int j = 0;
    int money = weekn + 1;
    while (j < day) {
      res += money;
      money++;
      j++;
    }

    if (weekn >= 1) {
      int i = 1;
      while (i <= weekn) {
        res += sumSeven(i);
        i++;
      }
    }

    return res;
  }

  int sumSeven(int s) {
    int sum = 0;
    int n = 1;
    while (n < =7) {
      sum += s;
      s++;
      n++;
    }
    return sum;
  }
};
